### PR TITLE
Global Adapter related Implementation

### LINKED ISSUE #2106: Connecting to Global Adapter and fetch APIs via XDS
### Describe your problem(s)
For the choreo deployment, the global adapter will be the component that provides information on which APIs to pull. For that it is required to implement XDS client within adapter component.

### How will you implement it

---

### PR DESCRIPTION
**Pull request by VirajSalaka**
### Purpose
This includes the connecting to the global adapter and fetching the APIs from global adapter.

### Issues

Fixes #2106 
Fixes part of https://github.com/wso2-enterprise/choreo/issues/4332

### Automation tests
 - Unit tests added: No
 - Integration tests added: No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [ ] Assigned 'Type' label
- [ ] Assigned the project
- [ ] Validated respective github issues
- [ ] Assigned milestone to the github issue(s)

### CODE REVIEW

**Comment by VirajSalaka** [comment]
Review was comments can be viewed in https://github.com/VirajSalaka/product-microgateway/pull/4

File: adapter/internal/ga/client.go [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+	"google.golang.org/genproto/googleapis/rpc/status"
+	"google.golang.org/grpc"
+)
+
+var (
+	// apiRevision Map keeps apiUUID -> revisionUUID. This is used only for the communication between global adapter and adapter
+	// The purpose here is to identify if the certain API's revision is already added to the XDS cache.
+	apiRevisionMap map[string]string
+	// Last Acknowledged Response from the global adapter
+	laskAckedResponse *discovery.DiscoveryResponse
```
Comment (mpmunasinghe) [inline adapter/internal/ga/client.go:40]: Typo

File: adapter/internal/ga/client.go [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+}
+
+func init() {
+	apiRevisionMap = make(map[string]string)
+	laskAckedResponse = &discovery.DiscoveryResponse{}
+	GAAPIChannel = make(chan *APIEvent)
+}
+
+func initConnection(xdsURL string) error {
+	// TODO: (VirajSalaka) Close Connection
```
Comment (mpmunasinghe) [inline adapter/internal/ga/client.go:71]: What is the connection close condition?

**Review by mpmunasinghe — APPROVED** [review]

**Comment by VirajSalaka** [comment]
Closing the issue, since we do not need it for main branch at the moment. The PR for feature branch would remain open.

**Comment by VirajSalaka** [comment]
Closing the PR as this is supposed to be sent to the feature/global-adapter branch

File: adapter/internal/ga/client.go (hunk 2 above)
Comment (VirajSalaka) [reply adapter/internal/ga/client.go]: removed via https://github.com/wso2/product-microgateway/pull/2107/commits/13c129233dc5f1ab801d14f9d923d1a6bc2d8752
closing the connection would be required when retrying is implemented.

File: adapter/internal/ga/client.go (hunk 1 above)
Comment (VirajSalaka) [reply adapter/internal/ga/client.go]: fixed via https://github.com/wso2/product-microgateway/pull/2107/commits/13c129233dc5f1ab801d14f9d923d1a6bc2d8752

**Pull request closed without merging**